integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -591,6 +592,44 @@ object CarbonDataRDDFactory {
                      s"${ carbonLoadModel.getDatabaseName }.${ carbonLoadModel.getTableName }")
       }
 
+      // code to handle Pre-Priming cache for loading
+      val indexServerEnabled = CarbonProperties.getInstance().isDistributedPruningEnabled(
+        carbonLoadModel.getDatabaseName, carbonLoadModel.getTableName)
+      val prePrimingEnabled = CarbonProperties.getInstance().isIndexServerPrePrimingEnabled()
+      if (indexServerEnabled && prePrimingEnabled) {
+        val readCommittedScope = new TableStatusReadCommittedScope(AbsoluteTableIdentifier.from(
+          carbonLoadModel.getTablePath), hadoopConf)
+        if (carbonLoadModel.getSegmentId != null) {
+          val loadDetailsForCurrentSegment = readCommittedScope.getSegmentList
+            .find(_.getLoadName.equalsIgnoreCase(carbonLoadModel.getSegmentId)).get
+          val segmentToWrite: Segment = new Segment(carbonLoadModel.getSegmentId,
+            loadDetailsForCurrentSegment.getSegmentFile,
+            readCommittedScope, loadDetailsForCurrentSegment)
+          val invalidsegment: List[Nothing] = List()
+
+          if (carbonTable.isChildDataMap || carbonTable.isChildTable) {
+            LOGGER.info("Loading Child in cache")
+          }
+          else {
+            LOGGER.info("Loading main table in cache")
```

Comment:
in log, pringt the uniqtable name and segment id also, while loading to cache

Reply:
done